docs/adr/README.md

```diff
@@ -0,0 +1,2 @@
+# Architectural Decision Records
+This directory contains a series of Architectural Decision Records or "ADRs" for the uniffi project. We're going to try to use it as a kind of collective memory of the decisions we've made and the path we've taken to get the project to its current point.
```

Comment:
It would be good to add the links explaining what ADRs to the README so that info is easier to find for new folks.

https://github.com/bitcoindevkit/bdk-ffi/pull/412#issuecomment-1799138670